app/event.tsx

```diff
@@ -69,19 +72,20 @@ export default function ModalScreen() {
 					reservationToken: reservation.data.reservationToken,
 					placeId: reservation.data.placeId,
 				};
-				await AsyncStorage.setItem(event.id.toString(), JSON.stringify(reservationData))
+				await AsyncStorage.setItem(event.id.toString() + '_' + reservationData.placeId.toString(), JSON.stringify(reservationData))
 					.then(() => {
 						// console.log("Stored reservation: " + JSON.stringify(reservationData));
 					});
 
 				// refetch event
-				setLoading(true);
 				getEvent()
 					.then(() => {
 						setLoading(false);
+					})
+					.then(() => {
+						Alert.alert('Reservation successful');
 					});
```

Comment:
```suggestion
				getEvent()
					.then(() => {
						Alert.alert('Reservation successful');
					})
					.finally(() => {
						setLoading(false);
					});
```
If getEvent() fails, we don't want `loading` to stay forever.

Reply:
Done